lib/main.dart

```diff
@@ -1,78 +1,110 @@
 void main() {
-  runApp(const MyApp());
+  runApp(MyApp());
 }
 
 class MyApp extends StatelessWidget {
-  const MyApp({Key? key}) : super(key: key);
+  MyApp({Key? key}) : super(key: key);
 
   // This widget is the root of your application.
   @override
   Widget build(BuildContext context) {
     return FutureBuilder(
-        future: Firebase.initializeApp(
-            options: DefaultFirebaseOptions.currentPlatform),
-        builder: (context, snapshot) {
-          if (snapshot.connectionState == ConnectionState.waiting) {
-            return const CircularProgressIndicator();
-          } else {
-            return MaterialApp(
-              title: 'Meeting Scheduler',
-              theme: ThemeData(
-                primarySwatch: Colors.blue,
-              ),
-              home: const AuthGate(),
-            );
-          }
-        });
-  }
-}
-
-class AuthGate extends StatelessWidget {
-  const AuthGate({Key? key}) : super(key: key);
-
-  @override
-  Widget build(BuildContext context) {
-    return StreamBuilder<User?>(
-      stream: FirebaseAuth.instance.authStateChanges(),
+      future: Firebase.initializeApp(options: DefaultFirebaseOptions.currentPlatform),
       builder: (context, snapshot) {
-        return StreamBuilder<User?>(
-          stream: FirebaseAuth.instance.authStateChanges(),
-          builder: (context, snapshot) {
-            // User is not signed in
-            if (!snapshot.hasData) {
-              return const SignInScreen(providerConfigs: [
-                EmailProviderConfiguration(),
-                GoogleProviderConfiguration(
-                  clientId:
-                      '220727339122-b3fflgie9rq6b6atu8sfml3274snrs31.apps.googleusercontent.com',
-                ),
-              ]);
-            }
-
-            // Render your application if authenticated
-            return const HomePage();
-          },
-        );
+        if (snapshot.connectionState == ConnectionState.waiting) {
+          return const Center(
+            child: CircularProgressIndicator(),
+          );
+        } else {
+          return MaterialApp.router(
+            routeInformationParser: _router.routeInformationParser,
+            routerDelegate: _router.routerDelegate,
+            title: 'Meeting Scheduler',
+            theme: ThemeData(
+              primarySwatch: Colors.blue,
+            ),
+          );
+        }
       },
     );
   }
-}
-
-class HomePage extends StatelessWidget {
-  const HomePage({Key? key}) : super(key: key);
 
-  @override
-  Widget build(BuildContext context) {
-    return const Material(
-        child: Center(
-      child: Text('Hello'),
-    ));
-  }
+  final GoRouter _router = GoRouter(
+    urlPathStrategy: UrlPathStrategy.path,
+    routes: [
+      GoRoute(
+        path: '/',
+        name: 'auth',
+        pageBuilder: (context, state) => MaterialPage(
+          key: state.pageKey,
+          child: const AuthGate(),
+        ),
+      ),
+      // GoRoute(
```

Comment:
Why is this commented out?

Reply:
Because the AuthAate class is handling both the sign in page and the home page, we should probably discuss, because I think go_router should manage them individually, and be able to redirect to either of them, according to the auth status.

Reply:
Awesome, yeah I think there are potentially two things to discuss then:

- routing vs. building the home page and auth page
- whether or not we adopt "don't commit commented code" style

Let's chat on Discord and come back

Reply:
For reference, my suggestion is we adopt the same style as flutter/flutter, ie: 

https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-abandonware

"...we delete rather than commenting out code. Commented-out code will bitrot too fast to be useful, and will confuse people maintaining the code."

Reply:
Do we need to figure out the build widgets vs push pages question for this PR?  We don't really need to figure out the "no commented code" style choice either, but they're slightly different in the latter is a choice directly relevant to this PR.  If it feels like too much to bother with on this PR just sing out 🙂